components/region_cache_memory_engine/src/range_manager.rs

```diff
@@ -100,9 +100,11 @@ pub struct RangeManager {
     // further write batch being cached. We must ensure the cached write batch is empty at the time
     // the range becoming accessable range.
     //
-    // Note: the range transferred from pending_range *must be* performed by the peer whose region
-    // range equals to it. If split happened, the first noticed peer should first split the range
-    // in the pending_range and then only handles its part.
+    // Note: the region with range equaling to the range in the `pending_range` may have been
+    // split. This is fine, we just let the first child region that calls the prepare_for_apply
```

Comment:
Will children regions run concurrently and it may have race condition? That say the first thread calls  prepare_for_apply, but the other thread almost run at the same time and even be faster while first thread's calling prepare_for_apply

Reply:
That's fine because we have to lock before reading/updating meta in RangeManager. So if the second child calls prepare_for_apply early than the first child, and schedules the range to load, the first child will notices it's in loading status as it's range (as well as the second child) is within the range scheduled by the second child. 
Actually, it does not matter who schedules the task, whether it's range is within a loading range does matter.